File: erlang/secret-handshake/src/secret_handshake.erl
-module(secret_handshake).
-export([commands/1]).

commands(0) ->
    [];
commands(N) when N band 16 > 0 ->
    lists:reverse(commands(N - 16));
commands(N) when N band 1 > 0 ->
    ["wink" | commands(N - 1)];
commands(N) when N band 2 > 0 ->
    ["double blink" | commands(N - 2)];
commands(N) when N band 4 > 0 ->
    ["close your eyes" | commands(N - 4)];
commands(N) when N band 8 > 0 ->
    ["jump" | commands(N - 8)].

File: php/secret-handshake/SecretHandshake.php
<?php

class SecretHandshake {

    private const ACTIONS = [
        1 => 'wink',
        2 => 'double blink',
        4 => 'close your eyes',
        8 => 'jump',
    ];
    private const REVERSE = 16;

    public function commands(int $handshake): array {
        $actions = [];
        foreach (self::ACTIONS as $num => $action) {
            if ($num & $handshake) {
                $actions[] = $action;
            }
        }
        return $handshake & self::REVERSE ? array_reverse($actions) : $actions;
    }
}
